### ISSUE TITLE
Problem: client logic for syncing state stream is not implemented

### ISSUE THREAD

**Issue by yihuang** [crypto-org-chain/cronos#732]
sub-issue of https://github.com/crypto-org-chain/cronos/issues/725

- support sync file streamer files through both local file system and http static file server
- parse the files and feed to local versiondb.

### Open Questions

> could be left to separate issues.

- current sdk builtin file streamer store separate file for each tx, are there too many small files under single directories? can linux file system handle that?
  - alternative: merge messages into single file per block, create directories based on decimal prefix of block numbers.
- file streamer should write files atomically
  - just vendor and modify the file streamer implementation.
- client sync logic:
  - with single file per block, just download the block file one by one.
  - with multiple files per block, check the end-block file existence.

**Comment by yihuang** [comment]
## Components
- `versiondb`
  - save change sets to database.
  - support grpc query service
- `versiondb` streamer
  - listen to state changes and save change sets to versiondb.
- `file` streamer
  - listen to state changes and dump the data to local files, one file per block.
- `versiondb` local sync-er
  - can monitor `file` streamer output directory, read the changesets into `versiondb`.
- file synchronizers between remote machines [^1]
  - sync files between remote machines.

Several ways to wire things up:
1. sync semi-archive node
  - use `versiondb` streamer.
  - run grpc query service with `versiondb`.
2. async semi-archive node
  - use `file` streamer.
  - run versiondb local sync-er.
  - run grpc query service with `versiondb`.
3. remote semi-archive node
  - use `file` streamer
  - run [file syncer](https://github.com/lsyncd/lsyncd) to sync files to remote nodes
  - remote nodes run versiondb local sync-er
  - remote nodes run in grpc only mode, and run grpc query service with `versiondb`.

**Comment by mmsqe** [comment]
Actually we still need handle block file sequence, lsyncd seems only do rsync whatever is missing.

**Comment by mmsqe** [comment]
Yup, work like Git. I might just test with file watch 2 modes: sync via http -> feed db with synced block file

**Comment by JayT106** [comment]
Closing: this appears to already be resolved in current `main`.

Evidence: Superseded by the memiavl + versiondb architecture, which now serves historical/state-stream queries instead of the originally proposed file-streamer design.

_Automated triage pass over open issues; reopen if this is inaccurate._
